neasqc_wp61/data/data_processing/dataset_vectoriser.py

```diff
@@ -0,0 +1,82 @@
+import json
+import pandas as pd
+from transformers import BertTokenizer, BertModel
+import torch
+import numpy as np
+import argparse
+import os
+
+def vectorise(dataset, embedding):
+    '''
+    This function takes the path to a dataset and a choice of BERT embedding 
+    (sentence or word) as input and returns nothing. It then reads this dataset 
+    and from it, builds a corresponding dataset with the chosen embedding for 
+    each sentence.
+    '''
+    dftrain = pd.read_csv('./' + dataset, sep='\t', header=None)
+    dftrain = dftrain.drop(columns=[2])
+    dftrain.columns = ['class', 'sentence']
+    tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
+    bert_model = BertModel.from_pretrained('bert-base-uncased')
+
+    for param in bert_model.parameters():
+        param.requires_grad = False
+    bert_model.eval()
+
+
+       
+    if embedding == 'sentence':
+        print('Generating sentence embeddings. Please wait...')
+        sentence_embedding_list = []
+        for sentence in dftrain.sentence.values:
+            inputs = tokenizer.encode_plus(sentence).input_ids
+            inputs = torch.LongTensor(inputs)
+            #inputs = inputs.to(mps_device)
+            
+            with torch.no_grad():
+                sentence_embedding = bert_model(inputs.unsqueeze(0))[1].squeeze(0).cpu().detach().numpy()
+            sentence_embedding_list.append([tensor.item() for tensor in sentence_embedding])
+
+        dftrain['sentence_embedding'] = sentence_embedding_list
+        dftrain.to_csv(dataset.split('.')[0] + '_sentence_bert.csv', index=False)
```

Comment:
I've used this script with the command : 

python3.10 -m dataset_vectoriser ./../datasets/reduced_amazonreview_train.tsv -e sentence

It generates a dataset called _sentence_bert.csv in the same data_processing folder. It would be nice if it could place the dataset in the datasets folder, with name : reduced_amazonreview_train_sentence_bert.csv

Reply:
@pablo-lauret

Reply:
Fixed